### PR TITLE
Remove unnecessary `this->` expressions

### PR DESCRIPTION
**Pull request by AThousandShips**
Unnecessary and hurts readability IMO, also creates confusion over *why* someone would use this as it's redundant by default so when you look at the code you wonder why this was done, to me for example it makes it seem like it's accessing something in some other object, as you'd obviously never use `->` for the current object

Because we have shadowing warnings now any of these are absolutely redundant, as there can't be a method local variable with the same name

I suspect several of these were for previous shadowings without `p_` prefixes

### CODE REVIEW

File: modules/gdscript/gdscript_parser.h [hunk 1]
```diff
@@ -223,7 +223,7 @@ class GDScriptParser {
 		}
 
 		bool operator!=(const DataType &p_other) const {
-			return !(this->operator==(p_other));
+			return !(operator==(p_other));
```
Comment (AThousandShips) [inline modules/gdscript/gdscript_parser.h:226]:
```suggestion
			return !(*this == p_other);
```
Already added but will update after CI has run

**Review by fire — APPROVED** [review]
The pr is removing unnecessary `this->` and so it shouldn't break stuff. I scanned the code and it seemed to be entirely removing `this->`. Looks good to me.

**Comment by AThousandShips** [comment]
That's all the cases I could find by a search for it, save for the two fixed by:
* https://github.com/godotengine/godot/pull/87669

Also excluded cases like `p_this->` and `this->*` which are all valid

**Comment by lawnjelly** [comment]
To hazard a guess, some of these (e.g. `Rational128`) may have been copy pasted from templates, where the `this->` is necessary to compile on some compilers afaik.

**Comment by AThousandShips** [comment]
Will restore those just to be safe 🙂

**Comment by AThousandShips** [comment]
Added a few commented cases just for completeness

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by AThousandShips** [comment]
Thank you!

**Pull request merged**